### ISSUE TITLE
E3002 Unable to lint a CFT if CarrierGateway has a set of tags.

### ISSUE THREAD

**Issue by saehan-suh** [aws-cloudformation/cfn-python-lint#1755]
*cfn-lint version: 0.39.0

*Description of issue.*
I cannot lint a resource of type 'CarrierGateway' (AWS::EC2::CarrierGateway), if its properties have the tags.

Please provide as much information as possible:
* Template linting issues:
  If I try to lint a CFT with a CarrierGateway resource without tags, lint went well. But if I did include the tags in the resource, the `cfn-lint` returns the error message below and fails:
  ```
  E3002 Expecting an object at Resources/CarrierGateway/Properties/Tags
  ```

  Here's a code snippet that does work and does NOT work, respectively:
  ```yaml
    (...)
    # Linting successful
    CarrierGateway:
        Type: AWS::EC2::CarrierGateway
        Properties: 
            VpcId: !Ref myVPC
    (...)
   ```

  ```yaml
    (...)
    # Returns an E3002 error
    CarrierGateway:
        Type: AWS::EC2::CarrierGateway
        Properties: 
            VpcId: !Ref myVPC
            Tags:
            - Key: Name
              Value: My CAGW
    (...)
   ```

  * [CloudFormation - AWS::EC2::Carrier Gateway](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-carriergateway.html)
  * Validate if the issue still exists with the latest version of `cfn-lint` and/or the latest Spec files

**Comment by kddejong** [comment]
This one would be an easy patch.
